images/encryption/blockcipher.go

```diff
@@ -0,0 +1,96 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package encryption
```

Comment:
This package should be broken up such that the `encryption` package defines the interfaces primitives and the sub packages have the implementations. For example, you'd have `image/encryption`, `image/encryption/gpg` and `image/encryption/pkcs7` and so on.

Reply:
Changes for the next version.